### PR TITLE
Add more reliable absolute coordinates computation

### LINKED ISSUE #81: Find a reliable way to determining client-absolute coordinates of SModelElements
I realized that the command palette is placed incorrectly in different contexts. This can be reproduced, for instance, by removing the bootstrap css import from the `class-diagram.html` example.

This issue caused by the offsets of the parents of sprotty's base div so that the absolute positioning either takes the browser's page as their coordinate system's origin or sprotty's base div.

We should therefore find a computation of absolute positions of e.g. UI extensions that works for any HTML context the Sprotty base div might be placed in.

### PR DESCRIPTION
**Pull request by planger**
This computation is available in the function `getAbsoluteClientBounds` which is now also used for computing the absolute position of the command palette.

Fixes #81

### CODE REVIEW

File: src/features/command-palette/command-palette.ts [hunk 1]
```diff
@@ -82,16 +84,15 @@ export class CommandPalette extends AbstractUIExtension {
...
-            if (isBoundsAware(firstElement)) {
-                const viewport = findParentByFeature(firstElement, isViewport);
-                if (viewport) {
-                    x += (firstElement.bounds.x + firstElement.bounds.width - viewport.scroll.x) * viewport.zoom;
-                    y += (firstElement.bounds.y - viewport.scroll.y) * viewport.zoom;
-                } else {
-                    x += firstElement.bounds.x + firstElement.bounds.width;
-                    y += firstElement.bounds.y;
-                }
+            if (firstElement) {
```
Comment (spoenemann) [inline src/features/command-palette/command-palette.ts]: Why should `selectedElements[0]` ever be undefined?

File: src/features/bounds/model.ts [hunk 2]
```diff
@@ -96,6 +98,50 @@ export function getAbsoluteBounds(element: SModelElement): Bounds {
...
+    let height = 0;
+
+    const svgElementId = domHelper.createUniqueDOMElementId(element);
+    const svgElement = document.getElementById(svgElementId);
+    if (svgElement) {
+        const rect = svgElement.getBoundingClientRect();
+        const body = document.body;
+        const docElement = document.documentElement;
+        const scrollTop = window.pageYOffset || docElement.scrollTop || body.scrollTop;
+        const scrollLeft = window.pageXOffset || docElement.scrollLeft || body.scrollLeft;
```
Comment (spoenemann) [inline src/features/bounds/model.ts]: Why not use `window.scrollX` and `window.scrollY` as [documented on MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect)? Maybe then you wouldn't need to take into account the `clientTop` / `clientLeft` values.

File: src/features/bounds/model.ts [hunk 3]
```diff
@@ -96,6 +98,50 @@ export function getAbsoluteBounds(element: SModelElement): Bounds {
...
+        const clientTop = docElement.clientTop || body.clientTop || 0;
+        const clientLeft = docElement.clientLeft || body.clientLeft || 0;
+
+        x = rect.left + scrollLeft - clientLeft;
+        y = rect.top + scrollTop - clientTop;
+        width = rect.right - rect.left;
+        height = rect.bottom - rect.top;
```
Comment (spoenemann) [inline src/features/bounds/model.ts]: `rect.width` / `rect.height`

File: src/features/command-palette/command-palette.ts (hunk 1 above)
Comment (planger) [reply src/features/command-palette/command-palette.ts]: Oh, right, good catch. Thanks!

File: src/features/bounds/model.ts (hunk 2 above)
Comment (planger) [reply src/features/bounds/model.ts]: You're right, this seems to work too. Thanks!

**Comment by planger** [comment]
@spoenemann Thanks for your review, Miro! I force-pushed an updated commit addressing your feedback!

**Pull request merged**